decryptor/mysql/response_proxy.go

```diff
@@ -835,18 +641,18 @@ func (handler *Handler) QueryResponseHandler(ctx context.Context, packet *Packet
 					Debugln("Can't read packet with column description")
 				return err
 			}
-			output = append(output, fieldPacket)
 			if handler.expectEOFOnColumnDefinition() {
 				if fieldPacket.IsEOF() {
 					if i != fieldCount {
 						handler.logger.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorProtocolProcessing).Errorln("EOF and field count != current row packet count")
 						return ErrMalformPacket
 					}
+					output = append(output, fieldPacket)
 					break
 				}
 			}
 			handler.logger.WithField("column_index", i).Debugln("Parse field")
-			field, err := ParseResultField(fieldPacket.GetData())
+			field, err := ParseResultField(fieldPacket, handler.setting.TableSchemaStore())
```

Comment:
what about to split Parsing and updating type of field? or rename to something obvious that show the fact that field's type will be changed?

Reply:
Agree to rename. It looks like confusion